src/test/kotlin/test/ktortemplate/core/utils/pagination/TestTableExtensionsFilters.kt

```diff
@@ -0,0 +1,183 @@
+package test.ktortemplate.core.utils.pagination
+
+import io.ktor.server.testing.TestApplicationEngine
+import io.ktor.util.KtorExperimentalAPI
+import org.amshove.kluent.`should be equal to`
+import org.jetbrains.exposed.dao.id.EntityID
+import org.jetbrains.exposed.dao.id.IdTable
+import org.jetbrains.exposed.sql.Column
+import org.jetbrains.exposed.sql.SchemaUtils
+import org.jetbrains.exposed.sql.deleteAll
+import org.jetbrains.exposed.sql.insert
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.koin.core.inject
+import org.koin.test.KoinTest
+import org.testcontainers.junit.jupiter.Container
+import org.testcontainers.junit.jupiter.Testcontainers
+import test.ktortemplate.conf.database.DatabaseConnection
+import test.ktortemplate.containers.PgSQLContainerFactory
+import test.ktortemplate.core.testApp
+import java.util.UUID
+
+@KtorExperimentalAPI
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@Testcontainers
+class TestTableExtensionsFilters : KoinTest {
+
+    companion object {
+        @Container
+        private val dbContainer = PgSQLContainerFactory.newInstance()
+    }
+
+    private val dbc: DatabaseConnection by inject()
+
+    internal object TestTable : IdTable<String>("tests") {
+        override val id: Column<EntityID<String>> = varchar("id", 255).entityId()
+        val long = long("long")
+        val int = integer("int")
+        val string = varchar("string", 255)
+        val boolean = bool("boolean")
+        val uuid = uuid("uuid")
+
+        override val primaryKey = PrimaryKey(id)
+    }
+
+    private val testId: String = "testId"
+    private val testLong: Long = 1L
+    private val testInt: Int = 1
+    private val testString: String = "string"
+    private val testBoolean: Boolean = true
+    private val testUUID: UUID = UUID.randomUUID()
+
+    @AfterEach
+    fun cleanDatabase() {
+        dbc.query { TestTable.deleteAll() }
+    }
+
+    @Nested
+    inner class TestFilterId {
+        @Test
+        fun `String primary key type should be supported`() = testAppWithConfig {
```

Comment:
For this particular test, wouldn't it be simpler to start the koin context and test the database access stuff without the overhead of the ktor application? Same as we did on `TestCarService.kt`

Reply:
Yep thanks 👍